### PR TITLE
feat: Support `on_demand_throughput`

### PR DESCRIPTION
**Pull request by magreenbaum**
## Description
Support `on_demand_throughput`. 

## Motivation and Context
Closes: https://github.com/terraform-aws-modules/terraform-aws-dynamodb-table/issues/89

## Breaking Changes
No.

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -106,6 +115,15 @@ resource "aws_dynamodb_table" "this" {
     }
   }
 
+  dynamic "on_demand_throughput" {
+    for_each = length(var.on_demand_throughput) > 0 ? [var.on_demand_throughput] : []
+
+    content {
+      max_read_request_units  = try(on_demand_throughput.value.max_read_request_units, null)
```
Comment (v-greco) [inline main.tf:122]: Hello,

as per documentation the default value should be -1 and no "null". Please check on:

[on_demand_throughput](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dynamodb_table#on_demand_throughput)

**Review by v-greco — CHANGES_REQUESTED** [review]
Please check my comment about the default values ^_^

File: main.tf (hunk 1 above)
Comment (bryantbiggs) [reply main.tf:122]: no - @magreenbaum has written it correctly

**Review by bryantbiggs — APPROVED** [review]
thank you!

**Comment by bryantbiggs** [comment]
fixed through the GitHub UI and corrected the pull request settings to avoid similar errors in the future 
image

**Pull request merged**